ykllvmwrap/src/jitbuilder.cc

```diff
@@ -0,0 +1,298 @@
+using namespace llvm;
+using namespace std;
+// An atomic counter used to issue compiled traces with unique names.
+atomic<uint64_t> NextTraceIdx(0);
+
+#define TRACE_FUNC_PREFIX "__yk_compiled_trace_"
+#define YKTRACE_START "__yktrace_start_tracing"
+#define YKTRACE_STOP "__yktrace_stop_tracing"
+
+std::vector<Value *> getTraceInputs(Function *F, uintptr_t BBIdx) {
+  std::vector<Value *> Vec;
+  auto It = F->begin();
+  // Skip to the first block in the trace which contains the `start_tracing`
+  // call.
+  std::advance(It, BBIdx);
+  BasicBlock *BB = &*It;
+  bool found = false;
+  for (auto I = BB->begin(); I != BB->end(); I++) {
+    if (isa<CallInst>(I)) {
+      CallInst *CI = cast<CallInst>(&*I);
+      if (CI->getCalledFunction()->getName() == YKTRACE_START) {
+        // Skip first argument to start_tracing.
```

Comment:
Why do we skip it?

Reply:
It's just the tracing kind (hardware of software). We never want to include that stuff.

Reply:
OK, let's explain that (perhaps in another PR). [I can tell the first argument is skipped from the "+1" in the code; but I don't know why!]

Reply:
Why do I feel like this isn't the first time Laurie has commented on this? ;-)